test/e2e/major_upgrade_v3.go

```diff
@@ -97,12 +99,54 @@ func MajorUpgradeToV3(logger *log.Logger) error {
 				testnet.NoError("failed to get header", err)
 				if resp.Header.Version.App == v3.Version {
 					upgradeComplete = true
+					if upgradedHeight == 0 {
+						upgradedHeight = resp.Header.Height
+					}
```

Comment:
If you wanted to be extra sure you could make the assertion that no other node has a different upgrade height (i.e. all nodes did actually upgrade at the same height)